### DISCUSSION TITLE
v2.2.0

### DISCUSSION THREAD

**Discussion by brianch** [brianch/offline-chess-puzzles#11] [Announcements]
As usual, you need to download the lichess puzzle database [from here](https://database.lichess.org/#puzzles) and unpack it to the  "puzzles" directory of the app.

## What's new:
- Engine support on analysis
- Flip board (to practice seeing what's being threatened against you) #9 
- The last search settings are remembered when you open the app again #8 
- No more need to set the board size manually, just resize the window and it will adjust as expected (the size is also remembered the next time you open the app) 
- Improved color themes
- No need to restart the app after changing settings.
- Fix the FEN on the puzzle tab, it was from the position before the opponent's move was made (I wonder if anyone had noticed this)
- Added a "copy to clipboard" button on the FEN and links.

## Other Changes:
- There's no more 2 executables, the one that uses opengl has all the features we need and better compatibility, so we're only using it (Thanks for the development in the [iced gui library](https://github.com/iced-rs/iced)). The next version will probably use the software renderer they are working on, which should improve compatibility further.
- In some systems people needed to manually install the vcruntime140 dll before, this should be fixed now.  #3
- A few minor tweaks on the enable/disable behaviour of the buttons below the board, to make it more consistent

The Linux binaries were tested on Arch Linux and Linux Mint 21.

<hr /><em>This discussion was created from the release <a href='https://github.com/brianch/offline-chess-puzzles/releases/tag/v2.2.0'>v2.2.0</a>.</em>

**Comment by PanirPakora** [comment]
In the previuous release underpromotions were not allowed to be performed on the chessboard (as in https://lichess.org/training/E4XWH), I still have to install the 2.2.0 and I hope the issue has been solved. If not, it would be certainly something to put on  the TODO list. Thanks and congratulatiuon for giving chess enthusiasts the possibility to enjoy such a bounty of chesspuzzles!

**Reply by brianch** [reply]
Hi! Underpromotion works, you just need to select the piece first (there in the bottom of the search tab) before moving the pawn. To try it out right now you can select the "underpromotion" theme.

It has been like that since the start if I recall correctly,  but in the early versions it was a bit more counter-intuitive because the current selected promotion piece wasn't being highlighted in any way.

**Comment by PanirPakora** [comment]
Ok, tx! my mistake.



brianch ***@***.***> ezt írta (időpont: 2023. márc. 17., P
12:57):
